### ISSUE TITLE
How long before LTS date should we cut off V8 updates

### ISSUE THREAD

**Issue by MylesBorins** [nodejs/Release#246]
This has been discussed in the past and consensus was reached for 2 months.

Is this amount of time a good length?

**Comment by natorion** [comment]
V8 releases a new major version every ~ 6 weeks so 2 months would be a sensible choice. Maybe a range would better reflect your intention?

**Comment by bnoordhuis** [comment]
For a different perspective: as long as it's possible to update V8 without breaking backwards compatibility, is there a reason not to?  Too much work?

**Comment by MylesBorins** [comment]
@bnoordhuis I tend to agree with you there. The primary argument against was potentially introducing new regressions or changes to the performance profile.

**Comment by mhdawson** [comment]
That was the motivation for this discussion.  Many of us see a V8 release as many lines of change which may come with unforeseen consequences, particularly on the performance side.

It was mentioned that we had previous agreement for 2 months in terms of when to stop taking new V8 levels. @rvagg was going to dig up some of the past history on that one.

**Comment by bnoordhuis** [comment]
Do you agree bug fixes > performance?  There are a number of moderately severe V8 bugs in Node 4 that we won't fix because it's simply too much work.  Sooner or later, the same thing is going to happen with Node 6.

As well, any bugs we do fix won't get the same rigorous testing (performance-wise or otherwise) as they do upstream.

By the time we upgrade V8 in a LTS branch, it has had sufficient real-world coverage in Current that we know what its performance characteristics are.  We can make the decision on a case-by-case basis.

To summarize: upgrading > not upgrading.

**Comment by MylesBorins** [comment]
So I want to throw something out there... currently we have been updating other dependencies throughout LTS including libUV, zlib, ICU, and openssl. These have generally come in minor updates.

This has generally proven to be a boon for LTS, providing a better experience overall. There have been regressions that have been introduced by updating almost all of these dependencies, but we have in general been able to catch them quickly, especially since the updates had originally landed in Current before.

So is it completely out of band to suggest that we consider on going V8 udpates during LTS in minor releases if we are able to keep the ABI compatible?

**Comment by addaleax** [comment]
> So is it completely out of band to suggest that we consider on going V8 udpates during LTS in minor releases if we are able to keep the ABI compatible?

To be fair, V8 is vastly more complex and has, as I understand it, a completely different version support model than any of our other dependencies… I don’t want to play Devil’s Advocate and I do think the current situation is a good example of when we *should* upgrade V8 – at the same time I wouldn’t feel comfortable with making V8 bumps in LTS a regular occurrence.

**Comment by gibfahn** [comment]
>So is it completely out of band to suggest that we consider ongoing V8 update during LTS in minor releases if we are able to keep the ABI compatible?

Doesn't seem unreasonable to update V8 in a release, AFAICT it's all a question of how confident we are that it's a net improvement over the previous version.

>To be fair, V8 is vastly more complex and has, as I understand it, a completely different version support model than any of our other dependencies…

True, but it is also better tested (AIUI), and the parts of the api that are used in Chrome are well tested there. If the ABI shim is minimal, then in effect it's a semver-minor V8 update, and the only reason it's called a major bump is because V8 bump their major version every six weeks no matter what. I think it should be discussed on a case by case basis, but I don't think it's out of the question that we'd update in an LTS version.

**Comment by MylesBorins** [comment]
please reopen if there is more to discuss
